Source/PeripheralManager.swift

```diff
@@ -0,0 +1,119 @@
+    /// - warning: If you pass background queue to the method make sure to observe results on main thread for UI related code.
+    /// - parameter queue: Queue on which bluetooth callbacks are received. By default main thread is used.
+    /// - parameter options: An optional dictionary containing initialization options for a central manager.
+    /// For more info about it please refer to [Peripheral Manager initialization options](https://developer.apple.com/library/ios/documentation/CoreBluetooth/Reference/CBPeripheralManager_Class/index.html)
+    public convenience init(queue: DispatchQueue = .main,
+                            options: [String: Any]? = nil) {
+        self.init(peripheralManager: CBPeripheralManager(delegate: nil, queue: queue, options: options),
+                  queueScheduler: ConcurrentDispatchQueueScheduler(queue: queue))
+    }
+
+    // MARK: State
+
+    /// Current state of `PeripheralManager` instance described by `BluetoothState` which is equivalent to [CBManagerState](https://developer.apple.com/reference/corebluetooth/cbmanager/1648600-state).
+    /// - returns: Current state of `PeripheralManager` as `BluetoothState`.
+    public var state: BluetoothState {
+        return BluetoothState(rawValue: peripheralManager.state.rawValue) ?? .unsupported
+    }
+
+    /// Observable which emits next whenever state of `CentralManager` changes. Never completes and errors
+    public var stateChanges: Observable<BluetoothState> {
+        return delegateWrapper.rx_didUpdateState
+    }
+
+    public typealias AdvertisingStarted = Void
+    //TODO: Docs
+    public func startAdvertising(_ advertisementData: Variable<[String : Any]?>) -> Single<AdvertisingStarted> {
+        let observable = Observable<AdvertisingStarted>.create { [weak self] observer in
+            guard let strongSelf = self else {
+                observer.onError(BluetoothError.destroyed)
+                return Disposables.create()
+            }
+            if strongSelf.isAdvertising {
+                observer.onError(BluetoothError.peripheralAlreadyAdvertising)
+                return Disposables.create()
+            }
+            let disposable = strongSelf.delegateWrapper.rx_didStartAdvertising
+                .flatMap { error -> Observable<AdvertisingStarted> in
+                    if let error = error {
+                        throw BluetoothError.peripheralAdvertisingFailed(error)
+                    }
+                    return .just(())
+            }.subscribe(observer)
+            let advertisementChangesDisposable = advertisementData
+                .asObservable()
+                .skip(1)
+                .subscribe(onNext: { [weak self] advertisement in
+                    self?.peripheralManager.startAdvertising(advertisement)
+                })
+            strongSelf.peripheralManager.startAdvertising(advertisementData.value)
```

Comment:
Remove this line and skip(1) from above?

Reply:
Good catch! Thanks!